internal/auth/service_account.go

```diff
@@ -26,40 +27,41 @@ func ReconcileServiceAccount(er record.EventRecorder, k8sClient client.Client, n
 		if serviceAccount, err = reconcile.ServiceAccount(er, k8sClient, serviceAccount); err != nil {
 			return err
 		}
-		return reconcileServiceAccountTokenSecret(serviceAccount, k8sClient, namespace, resNames.ServiceAccountTokenSecret, owner)
+		_, err = ReconcileServiceAccountTokenSecret(serviceAccount, k8sClient, namespace, resNames.ServiceAccountTokenSecret, owner)
+		return err
 	}
 	return nil
 }
 
-func reconcileServiceAccountTokenSecret(serviceAccount *corev1.ServiceAccount, k8sClient client.Client, namespace, name string, owner metav1.OwnerReference) error {
-	desired := runtime.NewSecret(namespace, name, map[string][]byte{})
+func ReconcileServiceAccountTokenSecret(sa *corev1.ServiceAccount, k8sClient client.Client, namespace, name string, owner metav1.OwnerReference) (desired *corev1.Secret, err error) {
+
+	if sa == nil {
+		return nil, fmt.Errorf("serviceAccount not provided in-order to generate a token")
+	}
+
+	if sa.UID == "" {
+		if sa, err = serviceaccount.Get(k8sClient, sa.Namespace, sa.Name); err != nil {
```

Comment:
Why move "reading the ServiceAccount data" into this function instead of providing a full ServiceAccount into this method like on the existing call?